src/Core/Extractors/nNASOS.cs

```diff
@@ -0,0 +1,140 @@
+        // mistakenly tries to register as a plugin, crashing LaunchBox on startup.
+        string executablePath = Path.Combine(PathUtils.GetLaunchBoxNNASOSRootPath(), "nNASOS.exe");
+
+        public NNASOS()
+        {
+
+        }
+
+        public static bool SupportedType(string archivePath)
+        {
+            return PathUtils.HasExtension(archivePath, new string[] { ".7z", ".zip", ".iso.dec" });
+        }
+
+        public override bool Extract(string archivePath, string cachePath, string[] includeList = null, string[] excludeList = null)
+        {
+            string isoDecPath = string.Empty;
+            string extension = Path.GetExtension(archivePath).ToLower();
+            bool skipArchiveCheck = false;
+
+            if (Directory.Exists(cachePath))
+            {
+                string[] files = Directory.GetFiles(cachePath, "*.iso.dec");
+                if (files.Length > 0)
+                {
+                    // .iso.dec is already in the cache path, so no need to extract or copy anything
+                    // from archive path
+                    Logger.Log(string.Format("Skipping extract/copy from archive path as .iso.dec is already in cache path..."));
+                    isoDecPath = files[0];
+                    skipArchiveCheck = true;
+                }
+            }
+
+            if (!skipArchiveCheck)
+            {
+                if (extension.Equals(".7z") || extension.Equals(".zip"))
+                {
+                    Extractor zip = new Zip();
+                    if (zip.Extract(archivePath, cachePath, new string[] { "*.iso", "*.iso.dec" }, excludeList))
+                    {
+                        string[] files = Directory.GetFiles(cachePath, "*.iso");
+                        if (files.Length > 0)
+                        {
+                            // no need to further process with nNASOS as already in .iso format
+                            return true;
+                        }
+                        else
+                        {
+                            isoDecPath = Directory.GetFiles(cachePath, "*.iso.dec")[0];
+                        }
+                    }
+                }
+                else if (extension.Equals(".iso.dec"))
+                {
+                    Extractor robocopy = new Robocopy();
+                    if (robocopy.Extract(archivePath, cachePath, includeList, excludeList))
+                    {
+                        isoDecPath = Directory.GetFiles(cachePath, "*.iso.dec")[0];
+                    }
+                }
+                else if (extension.Equals(".iso"))
+                {
+                    Extractor robocopy = new Robocopy();
+                    if (robocopy.Extract(archivePath, cachePath, includeList, excludeList))
+                    {
+                        // no need to extract as already in .iso format
+                        return true;
+                    }
+                }
```

Comment:
This case won't be reached, as the extractor has only registered for .7z, .zip and (.iso).dec extensions in `SupportedType()`. Copying the .iso to the cache would be handled by the Extract or Copy option.
```suggestion
```